sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanInfo.scala

```diff
@@ -31,7 +31,6 @@ class SparkPlanInfo(
     val nodeName: String,
     val simpleString: String,
     val children: Seq[SparkPlanInfo],
-    val metadata: Map[String, String],
```

Comment:
This is a developer api and I don't think users can do anything useful with `metadata` because it was just a hack. should be safe to remove it.

Reply:
It seems that @LantaoJin brings back this for event loging at #22353 .